### PR TITLE
GitHub Actions: Add PHPStan

### PR DESCRIPTION
**Pull request by lippserd**

### CODE REVIEW

File: library/Icingadb/Widget/Detail/PerfDataTable.php [hunk 1]
```diff
@@ -88,7 +88,7 @@ public function assemble()
                 unset($keys[$key]);
                 continue;
             }
-            if (isset($col)) {
+            if (! empty($col)) {
```
Comment (yhabteab) [inline library/Icingadb/Widget/Detail/PerfDataTable.php:91]: **Before:**
Bildschirmfoto 2023-08-17 um 17 07 18


With your changes:
Bildschirmfoto 2023-08-17 um 17 06 49

File: library/Icingadb/Data/CsvResultSet.php [hunk 2]
```diff
@@ -14,6 +14,7 @@ class CsvResultSet extends ResultSet
 {
     protected $isCacheDisabled = true;
 
+    #[\ReturnTypeWillChange]
     public function current()
```
Comment (nilmerg) [inline library/Icingadb/Data/CsvResultSet.php:18]: :array ?

File: library/Icingadb/Data/JsonResultSet.php [hunk 3]
```diff
@@ -15,6 +15,7 @@ class JsonResultSet extends ResultSet
 {
     protected $isCacheDisabled = true;
 
+    #[\ReturnTypeWillChange]
     public function current()
```
Comment (nilmerg) [inline library/Icingadb/Data/JsonResultSet.php:19]: : array ?

File: library/Icingadb/Hook/ActionsHook/ObjectActionsHook.php [hunk 4]
```diff
@@ -40,11 +40,11 @@ final public static function loadActions(Model $object): HtmlElement
     {
         switch (true) {
             case $object instanceof Host:
-                /** @var HostActionsHook $hook */
+                /** @var HostActionsHook $hookName */
```
Comment (nilmerg) [inline library/Icingadb/Hook/ActionsHook/ObjectActionsHook.php:43]: phpstorm gets this right. This previously meant $hook for a reason, as that's what's used by the loop below.

File: library/Icingadb/Hook/ExtensionHook/ObjectDetailExtensionHook.php [hunk 5]
```diff
@@ -43,23 +43,23 @@ final public static function loadExtensions(Model $object): array
     {
         switch (true) {
             case $object instanceof Host:
-                /** @var HostDetailExtensionHook $hook */
+                /** @var HostDetailExtensionHook $hookName */
```
Comment (nilmerg) [inline library/Icingadb/Hook/ExtensionHook/ObjectDetailExtensionHook.php:46]: same here

File: library/Icingadb/Hook/ExtensionHook/ObjectsDetailExtensionHook.php [hunk 6]
```diff
@@ -39,11 +39,11 @@ final public static function loadExtensions(string $objectType, Query $query, Fi
     {
         switch ($objectType) {
             case 'host':
-                /** @var HostsDetailExtensionHook $hook */
```
Comment (nilmerg) [inline library/Icingadb/Hook/ExtensionHook/ObjectsDetailExtensionHook.php:42]: same here

File: library/Icingadb/Web/Controller.php [hunk 7]
```diff
@@ -428,7 +429,7 @@ protected function sendAsPdf()
         $time = DateFormatter::formatDateTime(time());
 
         $doc = (new PrintableHtmlDocument())
-            ->setTitle($this->view->title)
+            ->setTitle($this->view->title ?? '')
```
Comment (nilmerg) [inline library/Icingadb/Web/Controller.php:432]: is this really required?

File: library/Icingadb/Widget/Detail/PerfDataTable.php [hunk 8]
```diff
@@ -49,85 +46,70 @@ public function __construct(string $perfdataStr, int $limit = 0, string $color =
...
-                        new HtmlElement(
-                            'span',
-                            Attributes::create([
-                                'class' => ($value ? '' : 'no-value')
-                            ]),
-                            $value ? Text::create($value) : new EmptyState(t('None', 'value'))
-                        ),
-                        [ 'class' => ($column === 'label' ? 'title' : null) ]
+                        HtmlString::create($perfdata->asInlinePie($this->color)->render()),
+                        [ 'class' => 'sparkline-col']
```
Comment (nilmerg) [inline library/Icingadb/Widget/Detail/PerfDataTable.php:92]: please remove the leading space before 'class

**Review by nilmerg — CHANGES_REQUESTED** [review]

**Review by nilmerg — CHANGES_REQUESTED** [review]
please rebase and fix the conflict

**Review by nilmerg — APPROVED** [review]

**Pull request merged**